quadcontrol/control.c

```diff
@@ -174,18 +186,29 @@ static int quad_takeoff(const flight_mode_t *mode)
 
 static int quad_hover(const flight_mode_t *mode)
 {
-	time_t now;
+	time_t now, end, lastLog;
 	ekf_state_t state;
 
-	DEBUG_LOG("HOVER - alt: %d, time: %lld\n", mode->hover.alt, mode->hover.time);
+	log_enable();
+	log_print("HOVER - alt: %d, time: %lld\n", mode->hover.alt, mode->hover.time);
 
 	now = quad_timeMsGet();
+	end = now + mode->hover.time;
+	lastLog = 0;
 
 #if TEST_ATTITUDE
-	while (quad_timeMsGet() < now + quad_common.duration) {
+	while (now < end) {
+		if (now - lastLog > LOG_PERIOD) {
+			lastLog = now;
+			log_enable();
+		}
```

Comment:
To have `enable/disable` in one place;
```suggestion
               now = quad_timeMsGet();
		if (now - lastLog > LOG_PERIOD) {
			lastLog = now;
			log_enable();
		}
		else {
		        log_disable();
		}
```